editor-components/feature/feature12/feature12.tsx

```diff
@@ -88,22 +173,77 @@ class Feature12 extends BaseFeature {
   }
 
   render() {
+    const upperTitle = this.getPropValue("upperTitle");
+    const title = this.getPropValue("title");
+    const titleBackingText = this.getPropValue("titleBackingText");
+    const itemCount = this.getPropValue("itemCount");
+    const messageCards = this.castToObject<IMessages[]>("messageCard");
+    const firstItem = this.getPropValue("message");
+    const firstCardTitle = firstItem[0].value;
+    const buttonText = firstItem[1].value;
+    const backgroundImage = firstItem[2].value;
+   
     return (
-      <div className={this.decorateCSS("container")}>
-        <div className={this.decorateCSS("max-content")}>
+      <div className={this.decorateCSS("container")} style={{ paddingBottom: "0px" }}>
+        <div className={this.decorateCSS("upper-title")} style={{ fontSize: "1.2em", textAlign: "center" }}>
+          {upperTitle}
+        </div>
+        <div className={this.decorateCSS("shadow-header-title")} style={{ fontSize: "12.5em", textAlign: "center", marginBottom: "20px" }}>
+          {titleBackingText}
+        </div>
+        <div className={this.decorateCSS("header-title")} style={{ fontSize: "5em", textAlign: "center", marginTop: "35px" }}>
+          {title}
+        </div>
+        <div className={this.decorateCSS("max-content")} style={{ marginTop: "20px" }}>
           <div className={this.decorateCSS("content")}>
-            {this.castToObject<IMessages[]>("message-card").map(
-              (message: any, index: number) => (
-                <div className={this.decorateCSS("card-item-count")} style={{
-                  width: 90 / this.getPropValue("itemCount") + "%",
-                }}>
-                <div className={this.decorateCSS("message")} key={index}>
-                  <h3 className={this.decorateCSS("title")}>{message.title}</h3>
-                  <p className={this.decorateCSS("long-text")}>{message.description}</p>
-                </div>
-                </div>
-              )
+            {firstItem.length > 0 && (
+              <div
+                className={this.decorateCSS("card-item-first")}
+                style={{
+                  width: `${100 / itemCount}%`,
+                  backgroundImage: `linear-gradient(rgba(0, 0, 0, 0.7), rgba(0, 0, 0, 0.75)), url(${backgroundImage})`,
+                 
+                }}
+              >
+                <div className={this.decorateCSS("blurry-background")}></div>
+                <h3 className={this.decorateCSS("title")}>{firstCardTitle}</h3>
+                <button className={this.decorateCSS("button")}>
+                  {buttonText}
+                </button>
```

Comment:
don't render this button if buttonText is empty.
[see here](https://github.com/Teknodev/composer-tools/blob/main/editor-components/README.md#how-to-check-if-a-string-prop-is-empty)